### PR TITLE
Added ManeuverBuilder::Combine internal intersection unit tests

### PR DESCRIPTION
**Pull request by dgearhart**
@dnesbitt61 @kevinkreiser @gknisely

### CODE REVIEW

File: src/odin/enhancedtrippath.cc [hunk 1]
```diff
@@ -166,46 +159,135 @@ std::string EnhancedTripPath_Edge::ToString() const {
...
-        str += "/";
-      str += name;
-    }
+    str += ListToString(this->sign().exit_name());
+  }
+
+  return str;
+}
+
+std::string EnhancedTripPath_Edge::ToUnitTestString() const {
```
Comment (kevinkreiser) [inline src/odin/enhancedtrippath.cc]: i would really really appreciate it if, when you needed to test something and have to alter the class to do it, that you'd derive another class from this base in the test itself, and implement the extra test methods there

File: src/odin/enhancedtrippath.cc [hunk 2]
```diff
@@ -166,46 +159,135 @@ std::string EnhancedTripPath_Edge::ToString() const {
...
+    if (is_first)
+      is_first = false;
+    else
+      str += "/";
+    str += item;
+  }
+  return str;
+}
+
+std::string EnhancedTripPath_Edge::ListToUnitTestString(
```
Comment (kevinkreiser) [inline src/odin/enhancedtrippath.cc]: ditto above. maybe it doesnt work in this instance, but i think its cleaner to keep test code in the tests

**Comment by kevinkreiser** [comment]
other than general comments about putting test code only in the tests, looks fine to me

File: src/odin/enhancedtrippath.cc (hunk 1 above)
Comment (dgearhart) [inline src/odin/enhancedtrippath.cc]: These are not used in the tests - I will update the name. Sorry for the confusion

**Pull request merged**